awx/plugins/inventory/awxrest.py

```diff
@@ -1,183 +0,0 @@
-    def __init__(self, token):
-        self.token = token
-
-    def __call__(self, request):
-        request.headers['Authorization'] = 'Token %s' % self.token
-        return request
-
-class InventoryScript(object):
-
-    def __init__(self, **options):
-        self.options = options
-
-    def get_data(self, output):
-        parts = urlparse.urlsplit(self.base_url)
-        if parts.username and parts.password:
-            auth = (parts.username, parts.password)
-        elif self.auth_token:
-            auth = TokenAuth(self.auth_token)
-        else:
-            auth = None
-        port = parts.port or (443 if parts.scheme == 'https' else 80)
-        url = urlparse.urlunsplit([parts.scheme,
-                                   '%s:%d' % (parts.hostname, port),
-                                   parts.path, parts.query, parts.fragment])
-        url_path = '/api/v1/inventories/%d/script/' % self.inventory_id
-        q = {}
-        if self.show_all:
-            q['all'] = 1
-        if self.hostname:
-            q['host'] = self.hostname
-        elif self.hostvars:
-            q['hostvars'] = 1
-        url_path += '?%s' % urllib.urlencode(q)
-        url = urlparse.urljoin(url, url_path)
-        response = requests.get(url, auth=auth)
-        response.raise_for_status()
-        output.write(json.dumps(json.loads(response.content),
-                                    indent=self.indent) + '\n')
-
-    def run(self, output=sys.stdout):
-        try:
-            self.base_url = self.options.get('base_url', '') or \
-                os.getenv('REST_API_URL', '')
-            if not self.base_url:
-                raise ValueError('No REST API URL specified')
-            self.auth_token = self.options.get('authtoken', '') or \
-                os.getenv('REST_API_TOKEN', '')
-            parts = urlparse.urlsplit(self.base_url)
-            if not (parts.username and parts.password) and not self.auth_token:
-                raise ValueError('No username/password specified in REST API '
-                                 'URL, and no REST API token provided')
-            try:
-                # Command line argument takes precedence over environment
-                # variable.
-                self.inventory_id = int(self.options.get('inventory_id', 0) or
-                                        os.getenv('INVENTORY_ID', 0))
-            except ValueError:
-                raise ValueError('Inventory ID must be an integer')
-            if not self.inventory_id:
-                raise ValueError('No inventory ID specified')
-            self.hostname = self.options.get('hostname', '')
-            self.list_ = self.options.get('list', False)
-            self.hostvars = bool(self.options.get('hostvars', False) or
-                                 os.getenv('INVENTORY_HOSTVARS', ''))
-            self.show_all = bool(self.options.get('show_all', False) or
-                                 os.getenv('INVENTORY_ALL', ''))
```

Comment:
I think this shows:

```
self.show_all = bool(self.options.get('show_all', False) or
```

that the default value is False for calling awxrest.py.

See in tasks.py where you did

```
json_data = instance.inventory.get_script_data({}, hostvars=True, show_all=True)
```

I think you should set it to False, in order to not break the feature of host enablement. If the host isn't enabled, we don't want to run the playbook against it. Enablement is done by scripts, or can be toggled in the UI. Disabled hosts just can't have any jobs ran against them.

Reply:
Another good catch, it looks like the original behavior is *not* to specify `show_all`.

Reply:
My latest code fixes this; I tested by toggling enabled/disabled for a few hosts, and making sure they do/don't get used for playbook runs.